shoop/simple_cms/admin_module/views.py

```diff
@@ -69,6 +73,36 @@ def clean(self):
 
         return data
 
+    def validate_url(self, language_code, field_name, url):
+        """
+        Ensure URL given is unique.
+
+        Check through the pages translation model objects to make
+        sure that the url given doesn't already exist.
+
+        Possible failure cases:
+        for new page:
+        * URL already exists
+
+        for existing page:
+        * URL (other than owned by existing page) exists
+        * URL exists in other languages of existing page
+        """
+        qs = self._get_translation_model().objects.filter(url=url)
+
+        if not self.instance.pk:
+            if qs.exists():
+                self.add_error(field_name, _("URL already exists."))
```

Comment:
I don't really see how it's this function's responsibility to add the error too (also, this same line is copypasted thrice).